src/detect-engine.c

```diff
@@ -4679,6 +4680,13 @@ int DetectEngineReload(const SCInstance *suri)
     SCLogDebug("old_de_ctx should have been freed");
 
     SCLogNotice("rule reload complete");
+
+    /* The reload process potentially frees up large amounts of memory.
+     * Encourage the memory management system to reclaim as much as it
+     * can.
+     */
+    malloc_trim(0);
```

Comment:
we need a macro wrapper based on availability, it should live in util-mem.h

Man page says "This function is a GNU extension.", so we can't just assume that it is available as CI failures show.